### PR TITLE
Add some very light and jankey documentation

### PR DESCRIPTION
**Pull request by paultag**
This is almost completely untested, and mostly here as a base for us to work from.

### CODE REVIEW

File: docs/QUICKSTART.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+to work through that issue.
+
+This document assumes some base level understanding of the moving AWS parts
+involved, as well as go. This also assumes `hallow-cli` has been installed,
+and can be run on the command line, as well as the ability for your local
+computer to talk to the AWS API.
+
+## Push Hallow deployment zip to S3
+
+Pull the latest Hallow zip from GitHub actions, or build your own zip using
```
Comment (alex) [inline docs/QUICKSTART.md]:
```suggestion
Pull the latest Hallow zip from [GitHub Actions](https://github.com/hallowauth/hallow/actions?query=branch%3Amaster+event%3Apush), or build your own zip using
```

File: docs/QUICKSTART.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+need the bucket name, as well as the name of the file (should be `hallow.zip`).
+
+## Create Hallow in your AWS Environment
+
+First, we need to set up Hallow to run in your AWS environment. This will ask
+you where the bucket and file are, as well as which region you're in.
+
+```
+cd terraform
+terraform apply
```
Comment (alex) [inline docs/QUICKSTART.md]: Pretty sure this doesn't work, I'm almost positive you need to instantiate the module yourself.

File: docs/QUICKSTART.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+# and your endpoint is set to the right endpoint!
+
+export AWS_REGION=us-east-1
+export HALLOW_ENDPOINT=https://UNIQUE_ID.execute-api.REGION.amazonaws.com/prod/hallow
+```
+
+If you've never used ssh certificates before, there should be no keys
+when you run `ssh-add -L | grep cert`. If there are keys, you likely know
+how to debug the following steps on your own. If you don't feel OK debugging
+your ssh-agent, feel free to continue with the commands and if no breakage
```
Comment (alex) [inline docs/QUICKSTART.md]:
```suggestion
your `ssh-agent`, feel free to continue with the commands and if no breakage
```

File: docs/QUICKSTART.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
+# Hellow Quickstart
```
Comment (paultag) [inline docs/QUICKSTART.md]:
```suggestion
# Hallow Quickstart
```

File: docs/QUICKSTART.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+computer, so please be sure it's not important!
+
+#### Get our Principal Name
+
+First, let's figure out who the heck we are. We'll ask AWS who they think
+we are, to figure out who Hallow will think we are. Keep a note of this ARN,
+this will be used to configure the SSH daemon.
+
+```
+$ aws sts get-caller-identity | grep Arn
```
Comment (alex) [inline docs/QUICKSTART.md]:
```suggestion
$ aws sts get-caller-identity --query="Arn"
```

File: docs/QUICKSTART.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+Write the key from `hallow-ca.pub` into `/etc/ssh/hallow_cas` on the remote
+machine.
+
+Now, let's write out the AWS ARNs as the authorized principals for any
+users we would like to grant access to:
+
+```
+[user@remote-computer]$ sudo mkdir /etc/ssh/principals/
+[user@remote-computer]$ sudo echo arn::aws::iam::..........:root > /etc/ssh/principals/user
```
Comment (alex) [inline docs/QUICKSTART.md]:
```suggestion
[user@remote-computer]$ sudo echo arn::aws::iam::..........:root > /etc/ssh/principals/${USER}
```

Maybe?

File: docs/QUICKSTART.md (hunk 2 above)
Comment (paultag) [reply docs/QUICKSTART.md]: @alex I was able to deploy it just using `terraform apply` and filling out the questions :laughing: - want to update this bit, though?

Comment (alex) [reply docs/QUICKSTART.md]: Huh. Ok! Maybe it should say something like "for production deployments you'll probably want to instantiate this as a module"?

Comment (paultag) [reply docs/QUICKSTART.md]: @alex I can remove this if you want to put best practice in instead!

Comment (paultag) [reply docs/QUICKSTART.md]: (I'm only a few months into using TF, all my usage is entirely by accident)

Comment (alex) [reply docs/QUICKSTART.md]: Seems ok for a quickstart guide, I genuinely did not expect it to work. I think it'd be good to also have a quick reference to the fact that for real life you'll probably want something a tad fancier.

Comment (paultag) [reply docs/QUICKSTART.md]: @alex note added!

File: docs/QUICKSTART.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+#### Get our Principal Name
+
+First, let's figure out who the heck we are. We'll ask AWS who they think
+we are, to figure out who Hallow will think we are. Keep a note of this ARN,
+this will be used to configure the SSH daemon.
+
+```
+$ aws sts get-caller-identity --query="Arn"
+    "Arn": "arn:aws:iam::.......:root"
```
Comment (alex) [inline docs/QUICKSTART.md]:
```suggestion
"arn:aws:iam::.......:root"
```

**Pull request merged**